src/main/groovy/com/netgrif/application/engine/petrinet/domain/dataset/logic/action/ActionDelegate.groovy

```diff
@@ -1634,11 +1701,21 @@ class ActionDelegate {
              ])
          },
          uri           : { cl ->
-             item = workflowService.findOne(item.stringId)
              def uri = cl() as String
-             item.setUriNodeId(uriService.findByUri(uri).id)
-             workflowService.save(item)
+             if (item.stringId == useCase.stringId) {
+                 moveMenuItem(useCase, uri)
+             } else {
+                 moveMenuItem(workflowService.findOne(item.stringId), uri)
+             }
+         },
+         title         : { cl ->
+             item = workflowService.findOne(item.stringId)
```

Comment:
Why is `item` updated from DB only here and not in other branches of this function? `caseDefaultHeaders`, `taskDefaultHeaders` and `filter` use it without update.

Reply:
`item = workflowService.findOne(item.stringId)` was redundant since `item` is updated by `setData`. Other branches use `setData` as well or update the item before changing data